api/src/main/java/io/opencensus/trace/BaseMessageEvent.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.trace;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Superclass for {@link MessageEvent} and {@link NetworkEvent} to resolve API backward
+ * compatibility issue.
+ *
+ * @since 0.12
+ */
+// public for io.opencensus.trace.export.SpanData.
+@SuppressWarnings("deprecation")
+public abstract class BaseMessageEvent {
+  /**
+   * Cast or convert a {@link BaseMessageEvent} to {@link MessageEvent}.
+   *
+   * <p>Warning: if the input is a {@code NetworkEvent} and contains {@code kernelTimestamp}
+   * information, this information will be dropped.
+   *
+   * @param event the {@code BaseMessageEvent} that is being cast or converted.
+   * @return a {@code MessageEvent} representation of the input.
+   * @since 0.12
+   */
+  public static MessageEvent asMessageEvent(BaseMessageEvent event) {
```

Comment:
I think that the two static methods in this class could go in a utility class in the `internal` package, since they are only used by opencensus-api.

Reply:
Please move it to the internal package if they are not really public.

Reply:
What about moving the whole class to the internal package?

Reply:
I just move the class out from the package :)
Now I am moving the covert logic back in internal, and leaves the class itself in `io.opencensus.trace` as a part of the API.